### PR TITLE
[Tooling] Update fastlane + remove workaround for match mutation

### PR DESCRIPTION
**Pull request by AliSoftware**
## What

 - Update `fastlane` to `2.205.2`
 - Which finally allows us to remove the workaround / patch we had to use because of `match` mutating one of its parameter — as my fix for it in https://github.com/fastlane/fastlane/pull/20125 has finally been released in `2.205.2` 🎉 

## To Test

 - Run `bundle exec fastlane update_certs_and_profiles` and verify it doesn't crash

### CODE REVIEW

**Review by mokagio — APPROVED** [review]
🎉 

Yes! I've been checking for a Fastlane release daily waiting for your fix to ship. Glad it's finally here.

CircleCI failed with what looks like an error on their end. I restarted it hoping it was just a fluke.

File: Gemfile.lock [hunk 1]
```diff
@@ -369,4 +368,4 @@ DEPENDENCIES
   xcpretty-travis-formatter
 
 BUNDLED WITH
-   2.3.9
+   2.3.8
```
Comment (mokagio) [inline Gemfile.lock:371]: Nitpick.

I'd be good to only _upgrade_ the expected Bundler version. What do you think? 

I've never looked into it, but maybe there's a way to actually lock it. `bundle install` warns us if the version is below the required one, but I wonder if we can stop it from even moving on with the process. ...Adding it to the "things I'll look into when I'll have some time" list 🙃

Comment (AliSoftware) [reply Gemfile.lock:371]: Wow I'm actually surprised that the downgrade was allowed to happen here, good catch; I'd have expected my run of `bundle update fastlane` would have warned me about it at the very least 🤔 (or maybe it did and I was not attentive enough with the console output?)

I agree about only going up in versions; I'll upgrade my local Bundler version and re-run it to fix that; thanks!

Comment (AliSoftware) [reply Gemfile.lock:371]: Ah shoot just realized auto merge already kicked in 🙃
Not sure it's worth doing a PR _just_ for that minor version bump off Bundler in lock file now that this one has already landed then … 🤪

**Pull request merged**